<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="TransProR User Manual.">

<title>TransProRBook - 9&nbsp; Enrichment Spiralize</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Enrichment ggraph.html" rel="next">
<link href="./Enrichment ggtree.html" rel="prev">
<link href="./TransProR_rlogo.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="custom.css">
<meta property="og:title" content="TransProRBook - 9&nbsp; Enrichment Spiralize">
<meta property="og:description" content="R package TransProR.">
<meta property="og:image" content="https://github.com/SSSYDYSSS/TransProRBook/blob/main/TransProR_rlogo.png">
<meta property="og:site-name" content="TransProRBook">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Enrichment Spiralize.html"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Enrichment Spiralize</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./TransProR_rlogo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">TransProRBook</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/SSSYDYSSS/TransProRBook" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <div class="dropdown">
      <a href="" title="Share" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Share"><i class="bi bi-share"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
              <i class="bi bi-bi-twitter pe-1"></i>
            Twitter
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
              <i class="bi bi-bi-facebook pe-1"></i>
            Facebook
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
              <i class="bi bi-bi-linkedin pe-1"></i>
            LinkedIn
            </a>
          </li>
      </ul>
    </div>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Data Source and Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Data Source and Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Extract Rawdata from TCGA and GTEX.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Extract Rawdata from TCGA and GTEX</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Remove Batch Effect.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Remove Batch Effect</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Select DEGs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Select DEGs</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Comparison of Four Methods.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Comparison of Four Methods</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Comparison of Six Methods.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Comparison of Six Methods</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Merging TransProR and TransProPy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Merging TransProR and TransProPy</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Enrichment ggtree.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Enrichment ggtree</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Enrichment Spiralize.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Enrichment Spiralize</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Enrichment ggraph.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Enrichment ggraph</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Enrichment Circos Bar.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Enrichment Circos Bar</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./EnrichPolarBubble.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">New Enrichment Polar Bubble</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Circos Fruit.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Circos Fruit</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Enrichment Circlize.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Enrichment Circlize</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Contact.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Contact</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-data" id="toc-load-data" class="nav-link active" data-scroll-target="#load-data"><span class="header-section-number">9.1</span> load data</a></li>
  <li><a href="#convert-from-symbol-to-entrezid" id="toc-convert-from-symbol-to-entrezid" class="nav-link" data-scroll-target="#convert-from-symbol-to-entrezid"><span class="header-section-number">9.2</span> Convert from SYMBOL to ENTREZID</a></li>
  <li><a href="#set-color" id="toc-set-color" class="nav-link" data-scroll-target="#set-color"><span class="header-section-number">9.3</span> set color</a></li>
  <li><a href="#sort" id="toc-sort" class="nav-link" data-scroll-target="#sort"><span class="header-section-number">9.4</span> Sort</a></li>
  <li><a href="#random" id="toc-random" class="nav-link" data-scroll-target="#random"><span class="header-section-number">9.5</span> Random</a></li>
  <li><a href="#reference" id="toc-reference" class="nav-link" data-scroll-target="#reference"><span class="header-section-number">9.6</span> Reference</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/SSSYDYSSS/TransProRBook/edit/main/Enrichment Spiralize.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/SSSYDYSSS/TransProRBook/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/SSSYDYSSS/TransProRBook/blob/main/Enrichment Spiralize.qmd" class="toc-action">View source</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Enrichment Spiralize</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The spiral plots generated by this module provide an intuitive visual display, depicting the expression changes in different biological pathways across samples through gradient colors and spatial arrangement. This visualization method helps to more clearly identify and compare the activity differences in key pathways, thereby deepening the understanding of physiological mechanisms.</p>
</div>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TransProR)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="load-data" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="load-data"><span class="header-section-number">9.1</span> load data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../test_TransProR/generated_data1/removebatch_SKCM_Skin_TCGA_exp_tumor.rds"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>normal <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">'../test_TransProR/generated_data1/removebatch_SKCM_Skin_Normal_TCGA_GTEX_count.rds'</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge the datasets, ensuring both have genes as row names</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>all_count_exp <span class="ot">&lt;-</span> <span class="fu">merge</span>(tumor, normal, <span class="at">by =</span> <span class="st">"row.names"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>all_count_exp <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(all_count_exp, <span class="at">var =</span> <span class="st">"Row.names"</span>)  <span class="co"># Set the row names</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Drawing data</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># all_count_exp &lt;- log_transform(all_count_exp)</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>DEG_deseq2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">'../test_TransProR/Select DEGs/DEG_deseq2.Rdata'</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#head(all_count_exp, 1)</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DEG_deseq2, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          baseMean log2FoldChange      lfcSE      stat pvalue padj change
A1BG      134.6084      -2.549682 0.05677219 -44.91075      0    0   down
A2M     30208.9912       3.251168 0.08394645  38.72907      0    0     up
AADACL2   801.4538      -8.229956 0.18969649 -43.38486      0    0   down
AARS2    1153.5493       1.624753 0.03283522  49.48202      0    0 stable
AARSD1    567.8672      -2.082289 0.02275703 -91.50088      0    0 stable</code></pre>
</div>
</div>
</section>
<section id="convert-from-symbol-to-entrezid" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="convert-from-symbol-to-entrezid"><span class="header-section-number">9.2</span> Convert from SYMBOL to ENTREZID</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert from SYMBOL to ENTREZID for convenient enrichment analysis later. It's crucial to do this now as a direct conversion may result in a reduced set of genes due to non-one-to-one correspondence.</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># DEG_deseq2</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve gene list</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(DEG_deseq2)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform conversion</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">=</span> <span class="fu">bitr</span>(gene, <span class="at">fromType=</span><span class="st">"SYMBOL"</span>, <span class="at">toType=</span><span class="st">"ENTREZID"</span>, <span class="at">OrgDb=</span><span class="st">"org.Hs.eg.db"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bitr(gene, fromType = "SYMBOL", toType = "ENTREZID", OrgDb =
"org.Hs.eg.db"): 43.37% of input gene IDs are fail to map...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove duplicates and merge</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(gene, SYMBOL, <span class="at">.keep_all=</span><span class="cn">TRUE</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the SYMBOL column as a vector from the first dataset</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>symbols_vector <span class="ot">&lt;-</span> gene<span class="sc">$</span>SYMBOL</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the SYMBOL column to filter corresponding rows from the second dataset by row names</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>DEG_deseq2 <span class="ot">&lt;-</span> DEG_deseq2[<span class="fu">rownames</span>(DEG_deseq2) <span class="sc">%in%</span> symbols_vector, ]</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DEG_deseq2, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          baseMean log2FoldChange      lfcSE      stat pvalue padj change
A1BG      134.6084      -2.549682 0.05677219 -44.91075      0    0   down
A2M     30208.9912       3.251168 0.08394645  38.72907      0    0     up
AADACL2   801.4538      -8.229956 0.18969649 -43.38486      0    0   down
AARS2    1153.5493       1.624753 0.03283522  49.48202      0    0 stable
AARSD1    567.8672      -2.082289 0.02275703 -91.50088      0    0 stable</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>Diff_deseq2 <span class="ot">&lt;-</span> <span class="fu">filter_diff_genes</span>(</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  DEG_deseq2, </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_val_col =</span> <span class="st">"pvalue"</span>, </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">log_fc_col =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_val_threshold =</span> <span class="fl">0.05</span>, </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">log_fc_threshold =</span> <span class="dv">3</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># First, obtain a list of gene names from the row names of the first dataset</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Diff_deseq2)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the matching rows in the second dataframe</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>matched_rows <span class="ot">&lt;-</span> all_count_exp[gene_names, ]</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the mean for each row</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>averages <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(matched_rows, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Append the averages as a new column to the first dataframe</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>Diff_deseq2<span class="sc">$</span>average <span class="ot">&lt;-</span> averages</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>Diff_deseq2<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Diff_deseq2)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>Diff_deseq2<span class="sc">$</span>changetype <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(Diff_deseq2<span class="sc">$</span>change <span class="sc">==</span> <span class="st">'up'</span>, <span class="dv">1</span>, <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a small threshold value</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>small_value <span class="ot">&lt;-</span> .Machine<span class="sc">$</span>double.xmin</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Before calculating -log10, replace zeroes with the small threshold value and assign it to a new column</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>Diff_deseq2<span class="sc">$</span>log_pvalue <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(Diff_deseq2<span class="sc">$</span>pvalue <span class="sc">==</span> <span class="dv">0</span>, <span class="sc">-</span><span class="fu">log10</span>(small_value), <span class="sc">-</span><span class="fu">log10</span>(Diff_deseq2<span class="sc">$</span>pvalue))</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the expression data corresponding to the differentially expressed genes</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>heatdata_deseq2 <span class="ot">&lt;-</span> all_count_exp[<span class="fu">rownames</span>(Diff_deseq2), ]</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing heatdata for visualization</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>HeatdataDeseq2 <span class="ot">&lt;-</span> TransProR<span class="sc">::</span><span class="fu">process_heatdata</span>(heatdata_deseq2, </span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">selection =</span> <span class="dv">2</span>, </span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">custom_names =</span> <span class="cn">NULL</span>, </span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">num_names_per_group =</span> <span class="dv">3</span>, </span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">prefix_length =</span> <span class="dv">4</span>)</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>HeatdataDeseq2 <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(HeatdataDeseq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Using the msigdbr package to download and prepare for GSVA analysis with KEGG and GO gene sets</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="do">## KEGG</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>KEGG_df_all <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Homo sapiens"</span>, <span class="co"># Homo sapiens or Mus musculus</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">category =</span> <span class="st">"C2"</span>,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">subcategory =</span> <span class="st">"CP:KEGG"</span>) </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>KEGG_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(KEGG_df_all, gs_name, gs_exact_source, gene_symbol)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>kegg_list <span class="ot">&lt;-</span> <span class="fu">split</span>(KEGG_df<span class="sc">$</span>gene_symbol, KEGG_df<span class="sc">$</span>gs_name) <span class="co"># Grouping gene symbols by gs_name</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(kegg_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$KEGG_ABC_TRANSPORTERS
 [1] "ABCA1"  "ABCA10" "ABCA12" "ABCA13" "ABCA2"  "ABCA3"  "ABCA4"  "ABCA5" 
 [9] "ABCA6"  "ABCA7"  "ABCA8"  "ABCA9"  "ABCB1"  "ABCB10" "ABCB11" "ABCB11"
[17] "ABCB4"  "ABCB5"  "ABCB6"  "ABCB7"  "ABCB8"  "ABCB9"  "ABCC1"  "ABCC1" 
[25] "ABCC10" "ABCC11" "ABCC12" "ABCC2"  "ABCC3"  "ABCC4"  "ABCC5"  "ABCC6" 
[33] "ABCC6"  "ABCC8"  "ABCC9"  "ABCD1"  "ABCD2"  "ABCD3"  "ABCD4"  "ABCG1" 
[41] "ABCG2"  "ABCG4"  "ABCG5"  "ABCG8"  "CFTR"   "TAP1"   "TAP1"   "TAP1"  
[49] "TAP1"   "TAP1"   "TAP1"   "TAP1"   "TAP1"   "TAP2"   "TAP2"   "TAP2"  
[57] "TAP2"   "TAP2"   "TAP2"   "TAP2"   "TAP2"  

$KEGG_ACUTE_MYELOID_LEUKEMIA
 [1] "AKT1"     "AKT2"     "AKT3"     "AKT3"     "ARAF"     "BAD"     
 [7] "BRAF"     "CCNA1"    "CCND1"    "CEBPA"    "CHUK"     "EIF4EBP1"
[13] "FLT3"     "GRB2"     "HRAS"     "HRAS"     "IKBKB"    "IKBKG"   
[19] "JUP"      "KIT"      "KRAS"     "LEF1"     "MAP2K1"   "MAP2K2"  
[25] "MAPK1"    "MAPK3"    "MTOR"     "MYC"      "NFKB1"    "NRAS"    
[31] "PIK3CA"   "PIK3CB"   "PIK3CD"   "PIK3CG"   "PIK3R1"   "PIK3R2"  
[37] "PIK3R3"   "PIK3R5"   "PIM1"     "PIM2"     "PML"      "PPARD"   
[43] "RAF1"     "RARA"     "RELA"     "RPS6KB1"  "RPS6KB2"  "RUNX1"   
[49] "RUNX1T1"  "SOS1"     "SOS2"     "SPI1"     "STAT3"    "STAT5A"  
[55] "STAT5B"   "TCF7"     "TCF7L1"   "TCF7L2"   "ZBTB16"  

$KEGG_ADHERENS_JUNCTION
 [1] "ACP1"    "ACTB"    "ACTG1"   "ACTN1"   "ACTN2"   "ACTN3"   "ACTN4"  
 [8] "ACTN4"   "AFDN"    "BAIAP2"  "CDC42"   "CDH1"    "CREBBP"  "CSNK2A1"
[15] "CSNK2A2" "CSNK2B"  "CSNK2B"  "CSNK2B"  "CSNK2B"  "CSNK2B"  "CSNK2B" 
[22] "CSNK2B"  "CSNK2B"  "CTNNA1"  "CTNNA2"  "CTNNA3"  "CTNNB1"  "CTNND1" 
[29] "EGFR"    "EP300"   "ERBB2"   "FARP2"   "FER"     "FGFR1"   "FYN"    
[36] "IGF1R"   "INSR"    "IQGAP1"  "LEF1"    "LMO7"    "MAP3K7"  "MAPK1"  
[43] "MAPK3"   "MET"     "NECTIN1" "NECTIN2" "NECTIN3" "NECTIN4" "NLK"    
[50] "PARD3"   "PTPN1"   "PTPN6"   "PTPRB"   "PTPRF"   "PTPRJ"   "PTPRM"  
[57] "RAC1"    "RAC2"    "RAC3"    "RHOA"    "SMAD2"   "SMAD3"   "SMAD4"  
[64] "SNAI1"   "SNAI2"   "SORBS1"  "SRC"     "SSX2IP"  "TCF7"    "TCF7L1" 
[71] "TCF7L2"  "TGFBR1"  "TGFBR2"  "TJP1"    "TJP1"    "VCL"     "WAS"    
[78] "WASF1"   "WASF2"   "WASF3"   "WASL"    "YES1"   

$KEGG_ADIPOCYTOKINE_SIGNALING_PATHWAY
 [1] "ACACB"    "ACSL1"    "ACSL3"    "ACSL4"    "ACSL5"    "ACSL6"   
 [7] "ADIPOQ"   "ADIPOR1"  "ADIPOR2"  "ADIPOR2"  "AGRP"     "AKT1"    
[13] "AKT2"     "AKT3"     "AKT3"     "CAMKK1"   "CAMKK2"   "CD36"    
[19] "CHUK"     "CPT1A"    "CPT1B"    "CPT1C"    "G6PC1"    "G6PC2"   
[25] "G6PC2"    "IKBKB"    "IKBKG"    "IRS1"     "IRS2"     "IRS4"    
[31] "JAK2"     "LEP"      "LEPR"     "MAPK10"   "MAPK8"    "MAPK9"   
[37] "MTOR"     "NFKB1"    "NFKBIA"   "NFKBIB"   "NFKBIB"   "NFKBIE"  
[43] "NPY"      "PCK1"     "PCK2"     "PCK2"     "POMC"     "PPARA"   
[49] "PPARGC1A" "PRKAA1"   "PRKAA2"   "PRKAB1"   "PRKAB2"   "PRKAG1"  
[55] "PRKAG2"   "PRKAG3"   "PRKCQ"    "PTPN11"   "RELA"     "RXRA"    
[61] "RXRB"     "RXRB"     "RXRB"     "RXRB"     "RXRB"     "RXRB"    
[67] "RXRG"     "SLC2A1"   "SLC2A4"   "SLC2A4"   "SOCS3"    "STAT3"   
[73] "STK11"    "TNF"      "TNF"      "TNF"      "TNF"      "TNF"     
[79] "TNF"      "TNF"      "TNF"      "TNFRSF1A" "TNFRSF1B" "TRADD"   
[85] "TRAF2"   

$KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM
 [1] "ABAT"    "ACY3"    "ADSL"    "ADSS1"   "ADSS2"   "AGXT"    "AGXT2"  
 [8] "ALDH4A1" "ALDH5A1" "ASL"     "ASNS"    "ASPA"    "ASS1"    "CAD"    
[15] "CPS1"    "DDO"     "GAD1"    "GAD2"    "GFPT1"   "GFPT2"   "GLS"    
[22] "GLS2"    "GLUD1"   "GLUD2"   "GLUD2"   "GLUL"    "GOT1"    "GOT2"   
[29] "GPT"     "GPT2"    "IL4I1"   "NIT2"    "PPAT"   

$KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION
 [1] "ATP1A1"   "ATP1A2"   "ATP1A3"   "ATP1A4"   "ATP1B1"   "ATP1B2"  
 [7] "ATP1B3"   "ATP1B4"   "FXYD2"    "FXYD4"    "HSD11B1"  "HSD11B2" 
[13] "IGF1"     "INS"      "INSR"     "IRS1"     "IRS2"     "IRS4"    
[19] "KCNJ1"    "KRAS"     "MAPK1"    "MAPK3"    "NEDD4L"   "NR3C2"   
[25] "PDPK1"    "PIK3CA"   "PIK3CB"   "PIK3CD"   "PIK3CG"   "PIK3R1"  
[31] "PIK3R2"   "PIK3R3"   "PIK3R5"   "PRKCA"    "PRKCB"    "PRKCG"   
[37] "SCNN1A"   "SCNN1B"   "SCNN1G"   "SFN"      "SGK1"     "SLC9A3R2"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## GO</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>GO_df_all <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Homo sapiens"</span>,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">category =</span> <span class="st">"C5"</span>)  </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>GO_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(GO_df_all, gs_name, gene_symbol, gs_exact_source, gs_subcat)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>GO_df <span class="ot">&lt;-</span> GO_df[GO_df<span class="sc">$</span>gs_subcat <span class="sc">!=</span> <span class="st">"HPO"</span>,]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>go_list <span class="ot">&lt;-</span> <span class="fu">split</span>(GO_df<span class="sc">$</span>gene_symbol, GO_df<span class="sc">$</span>gs_name) <span class="co"># Grouping gene symbols by gs_name</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(go_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS
[1] "AASDHPPT" "ALDH1L1"  "ALDH1L2"  "MTHFD1"   "MTHFD1L"  "MTHFD2L" 

$GOBP_2_OXOGLUTARATE_METABOLIC_PROCESS
 [1] "AADAT"  "ADHFE1" "D2HGDH" "DLST"   "GOT1"   "GOT2"   "GPT2"   "IDH1"  
 [9] "IDH2"   "IDH3B"  "KYAT3"  "L2HGDH" "MRPS36" "MRPS36" "OGDH"   "OGDHL" 
[17] "PHYH"   "TAT"   

$GOBP_2FE_2S_CLUSTER_ASSEMBLY
[1] "BOLA2"  "BOLA2B" "GLRX3"  "GLRX5"  "HSCB"   "NFS1"  

$GOBP_3_PHOSPHOADENOSINE_5_PHOSPHOSULFATE_BIOSYNTHETIC_PROCESS
[1] "PAPSS1"  "PAPSS2"  "SLC26A1" "SLC26A2" "SLC35B2" "SLC35B3"

$GOBP_3_PHOSPHOADENOSINE_5_PHOSPHOSULFATE_METABOLIC_PROCESS
 [1] "ABHD14B" "BPNT1"   "ENPP1"   "PAPSS1"  "PAPSS2"  "SLC26A1" "SLC26A2"
 [8] "SLC35B2" "SLC35B3" "SULT1A1" "SULT1A2" "SULT1A3" "SULT1A4" "SULT1B1"
[15] "SULT1C3" "SULT1C4" "SULT1E1" "SULT2A1" "SULT2B1" "TPST1"   "TPST2"  

$GOBP_3_UTR_MEDIATED_MRNA_DESTABILIZATION
 [1] "CPEB3"   "DHX36"   "DHX36"   "DND1"    "DND1"    "HNRNPD"  "KHSRP"  
 [8] "MOV10"   "PLEKHN1" "RBM24"   "RC3H1"   "TARDBP"  "TRIM71"  "UPF1"   
[15] "ZC3H12A" "ZC3H12D" "ZFP36"   "ZFP36L1" "ZFP36L2"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(gsva_mat, "gsva_go_matrix.csv")</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>ssgsea_kegg <span class="ot">&lt;-</span> <span class="fu">gsva</span>(<span class="at">expr =</span> HeatdataDeseq2, </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gset.idx.list =</span> kegg_list, </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">kcdf =</span> <span class="st">"Poisson"</span>, <span class="co">#"Gaussian" for logCPM, logRPKM, logTPM, "Poisson" for counts</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"ssgsea"</span>,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#parallel.sz = parallel::detectCores() # Utilize all available cores</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Calling gsva(expr=., gset.idx.list=., method=., ...) is deprecated;
use a method-specific parameter object (see '?gsva').</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .filterFeatures(expr, method): 37 genes with constant expression
values throughout the samples.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .gsva(expr, mapped.gset.idx.list, method, kcdf, rnaseq, abs.ranking,
: Some gene sets have size one. Consider setting 'min.sz &gt; 1'.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Estimating ssGSEA scores for 171 gene sets.
[1] "Calculating ranks..."
[1] "Calculating absolute values from ranks..."

  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |======================================================================| 100%

[1] "Normalizing..."</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>ssgsea_kegg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ssgsea_kegg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-color" class="level2" data-number="9.3">
<h2 data-number="9.3" class="anchored" data-anchor-id="set-color"><span class="header-section-number">9.3</span> set color</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>nTop <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>results1 <span class="ot">&lt;-</span> <span class="fu">extract_ntop_pathways</span>(<span class="at">ssgsea_kegg =</span> ssgsea_kegg, <span class="at">nTop =</span> nTop)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>results2 <span class="ot">&lt;-</span> <span class="fu">extract_positive_pathways</span>(ssgsea_kegg, <span class="at">max_paths_per_sample =</span> <span class="dv">5</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the required number of transition colors</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>number_of_colors <span class="ot">&lt;-</span> nTop</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Define start and end colors</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>start_color1 <span class="ot">&lt;-</span> <span class="st">"#fd8d63"</span>  </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>end_color1 <span class="ot">&lt;-</span> <span class="st">"#ededed"</span>    </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Use colorRampPalette to generate color transitions</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>color_palette1 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color1, end_color1))</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>interpolated_colors1 <span class="ot">&lt;-</span> <span class="fu">color_palette1</span>(number_of_colors)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>start_color2 <span class="ot">&lt;-</span> <span class="st">"#673ab7"</span>  </span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>end_color2 <span class="ot">&lt;-</span> <span class="st">"#d1c4e9"</span>    </span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>color_palette2 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color2, end_color2))</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>interpolated_colors2 <span class="ot">&lt;-</span> <span class="fu">color_palette2</span>(number_of_colors)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>start_color3 <span class="ot">&lt;-</span> <span class="st">"#6ac1a7"</span>  </span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>end_color3 <span class="ot">&lt;-</span> <span class="st">"#ededed"</span>    </span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>color_palette3 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color3, end_color3))</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>interpolated_colors3 <span class="ot">&lt;-</span> <span class="fu">color_palette3</span>(number_of_colors)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>start_color4 <span class="ot">&lt;-</span> <span class="st">"#2196f3"</span>  </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>end_color4 <span class="ot">&lt;-</span> <span class="st">"#bbdefb"</span>    </span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>color_palette4 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color4, end_color4))</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>interpolated_colors4 <span class="ot">&lt;-</span> <span class="fu">color_palette4</span>(number_of_colors)</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>start_color5 <span class="ot">&lt;-</span> <span class="st">"#ffdb37"</span>  </span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>end_color5 <span class="ot">&lt;-</span> <span class="st">"#ededed"</span>    </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>color_palette5 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color5, end_color5))</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>interpolated_colors5 <span class="ot">&lt;-</span> <span class="fu">color_palette5</span>(number_of_colors)</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>start_color6 <span class="ot">&lt;-</span> <span class="st">"#00bcd4"</span>  </span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>end_color6 <span class="ot">&lt;-</span> <span class="st">"#b2ebf2"</span>    </span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>color_palette6 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color6, end_color6))</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>interpolated_colors6 <span class="ot">&lt;-</span> <span class="fu">color_palette6</span>(number_of_colors)</span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>start_color7 <span class="ot">&lt;-</span> <span class="st">"#a7da55"</span>  </span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>end_color7 <span class="ot">&lt;-</span> <span class="st">"#ededed"</span>    </span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a>color_palette7 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color7, end_color7))</span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a>interpolated_colors7 <span class="ot">&lt;-</span> <span class="fu">color_palette7</span>(number_of_colors)</span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true" tabindex="-1"></a>start_color8 <span class="ot">&lt;-</span> <span class="st">"#43a047"</span>  </span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true" tabindex="-1"></a>end_color8 <span class="ot">&lt;-</span> <span class="st">"#c8e6c9"</span>    </span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true" tabindex="-1"></a>color_palette8 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color8, end_color8))</span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true" tabindex="-1"></a>interpolated_colors8 <span class="ot">&lt;-</span> <span class="fu">color_palette8</span>(number_of_colors)</span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true" tabindex="-1"></a>start_color9 <span class="ot">&lt;-</span> <span class="st">"#e68bc1"</span>  </span>
<span id="cb20-51"><a href="#cb20-51" aria-hidden="true" tabindex="-1"></a>end_color9 <span class="ot">&lt;-</span> <span class="st">"#ededed"</span>    </span>
<span id="cb20-52"><a href="#cb20-52" aria-hidden="true" tabindex="-1"></a>color_palette9 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color9, end_color9))</span>
<span id="cb20-53"><a href="#cb20-53" aria-hidden="true" tabindex="-1"></a>interpolated_colors9 <span class="ot">&lt;-</span> <span class="fu">color_palette9</span>(number_of_colors)</span>
<span id="cb20-54"><a href="#cb20-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-55"><a href="#cb20-55" aria-hidden="true" tabindex="-1"></a>start_color10 <span class="ot">&lt;-</span> <span class="st">"#cddc39"</span>  </span>
<span id="cb20-56"><a href="#cb20-56" aria-hidden="true" tabindex="-1"></a>end_color10 <span class="ot">&lt;-</span> <span class="st">"#f0f4c3"</span>    </span>
<span id="cb20-57"><a href="#cb20-57" aria-hidden="true" tabindex="-1"></a>color_palette10 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color10, end_color10))</span>
<span id="cb20-58"><a href="#cb20-58" aria-hidden="true" tabindex="-1"></a>interpolated_colors10 <span class="ot">&lt;-</span> <span class="fu">color_palette10</span>(number_of_colors)</span>
<span id="cb20-59"><a href="#cb20-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-60"><a href="#cb20-60" aria-hidden="true" tabindex="-1"></a>start_color11 <span class="ot">&lt;-</span> <span class="st">"#8fa1cc"</span>  </span>
<span id="cb20-61"><a href="#cb20-61" aria-hidden="true" tabindex="-1"></a>end_color11 <span class="ot">&lt;-</span> <span class="st">"#ededed"</span>    </span>
<span id="cb20-62"><a href="#cb20-62" aria-hidden="true" tabindex="-1"></a>color_palette11 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color11, end_color11))</span>
<span id="cb20-63"><a href="#cb20-63" aria-hidden="true" tabindex="-1"></a>interpolated_colors11 <span class="ot">&lt;-</span> <span class="fu">color_palette11</span>(number_of_colors)</span>
<span id="cb20-64"><a href="#cb20-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-65"><a href="#cb20-65" aria-hidden="true" tabindex="-1"></a>start_color12 <span class="ot">&lt;-</span> <span class="st">"#ff8f00"</span>  </span>
<span id="cb20-66"><a href="#cb20-66" aria-hidden="true" tabindex="-1"></a>end_color12 <span class="ot">&lt;-</span> <span class="st">"#ffecb3"</span>    </span>
<span id="cb20-67"><a href="#cb20-67" aria-hidden="true" tabindex="-1"></a>color_palette12 <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(start_color12, end_color12))</span>
<span id="cb20-68"><a href="#cb20-68" aria-hidden="true" tabindex="-1"></a>interpolated_colors12 <span class="ot">&lt;-</span> <span class="fu">color_palette12</span>(number_of_colors)</span>
<span id="cb20-69"><a href="#cb20-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-70"><a href="#cb20-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the resulting colors</span></span>
<span id="cb20-71"><a href="#cb20-71" aria-hidden="true" tabindex="-1"></a>Pathwaycolor2 <span class="ot">&lt;-</span> <span class="fu">c</span>(interpolated_colors1, interpolated_colors3, interpolated_colors5, interpolated_colors7, interpolated_colors9, interpolated_colors11)</span>
<span id="cb20-72"><a href="#cb20-72" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(Pathwaycolor2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "#FD8D63" "#F9A585" "#F5BDA8" "#F1D5CA" "#EDEDED" "#6AC1A7" "#8ACCB8"
 [8] "#ABD7CA" "#CCE2DB" "#EDEDED" "#FFDB37" "#FADF64" "#F6E392" "#F1E8BF"
[15] "#EDEDED" "#A7DA55" "#B8DE7B" "#CAE3A1" "#DBE8C7" "#EDEDED" "#E68BC1"
[22] "#E7A3CC" "#E9BCD7" "#EBD4E2" "#EDEDED" "#8FA1CC" "#A6B3D4" "#BEC7DC"
[29] "#D5DAE4" "#EDEDED"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>Samplecolor2 <span class="ot">&lt;-</span> <span class="fu">c</span>(start_color1, start_color3, start_color5, start_color7, start_color9, start_color11)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(Samplecolor2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "#fd8d63" "#6ac1a7" "#ffdb37" "#a7da55" "#e68bc1" "#8fa1cc"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>results1<span class="sc">$</span>PathwayColor <span class="ot">&lt;-</span> Pathwaycolor2</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>results2<span class="sc">$</span>PathwayColor <span class="ot">&lt;-</span> Pathwaycolor2</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a color mapping for the Sample column</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>Sample_colors2 <span class="ot">&lt;-</span> <span class="fu">setNames</span>(Samplecolor2, <span class="fu">unique</span>(results1<span class="sc">$</span>Sample))</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>Sample_colors2 <span class="ot">&lt;-</span> <span class="fu">setNames</span>(Samplecolor2, <span class="fu">unique</span>(results2<span class="sc">$</span>Sample))</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Add SampleColor column to the DataFrame</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>results1<span class="sc">$</span>SampleColor <span class="ot">&lt;-</span> Sample_colors2[results1<span class="sc">$</span>Sample]</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>results2<span class="sc">$</span>SampleColor <span class="ot">&lt;-</span> Sample_colors2[results2<span class="sc">$</span>Sample]</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results1))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                           Pathway Sample     Value
1                       KEGG_RIBOFLAVIN_METABOLISM TCGA_1 0.5417180
2                              KEGG_PRION_DISEASES TCGA_1 0.5324760
3                     KEGG_BETA_ALANINE_METABOLISM TCGA_1 0.5187272
4 KEGG_AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM TCGA_1 0.5169535
5                        KEGG_LEISHMANIA_INFECTION TCGA_1 0.5003448
6                       KEGG_RIBOFLAVIN_METABOLISM TCGA_2 0.5345860
  PathwayColor SampleColor
1      #FD8D63     #fd8d63
2      #F9A585     #fd8d63
3      #F5BDA8     #fd8d63
4      #F1D5CA     #fd8d63
5      #EDEDED     #fd8d63
6      #6AC1A7     #6ac1a7</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                     Pathway Sample     Value PathwayColor
1                    KEGG_ENDOMETRIAL_CANCER TCGA_1 0.4091193      #FD8D63
2       KEGG_ARGININE_AND_PROLINE_METABOLISM TCGA_1 0.1608353      #F9A585
3                 KEGG_RIBOFLAVIN_METABOLISM TCGA_1 0.5417180      #F5BDA8
4            KEGG_NON_HOMOLOGOUS_END_JOINING TCGA_1 0.2218331      #F1D5CA
5            KEGG_NON_SMALL_CELL_LUNG_CANCER TCGA_1 0.3934194      #EDEDED
6 KEGG_PATHOGENIC_ESCHERICHIA_COLI_INFECTION TCGA_2 0.1543440      #6AC1A7
  SampleColor
1     #fd8d63
2     #fd8d63
3     #fd8d63
4     #fd8d63
5     #fd8d63
6     #6ac1a7</code></pre>
</div>
</div>
</section>
<section id="sort" class="level2" data-number="9.4">
<h2 data-number="9.4" class="anchored" data-anchor-id="sort"><span class="header-section-number">9.4</span> Sort</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">enrichment_spiral_plots</span>(results1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Enrichment-Spiralize_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
</section>
<section id="random" class="level2" data-number="9.5">
<h2 data-number="9.5" class="anchored" data-anchor-id="random"><span class="header-section-number">9.5</span> Random</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">enrichment_spiral_plots</span>(results2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Enrichment-Spiralize_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
</section>
<section id="reference" class="level2" data-number="9.6">
<h2 data-number="9.6" class="anchored" data-anchor-id="reference"><span class="header-section-number">9.6</span> Reference</h2>
<blockquote class="blockquote">
<ul>
<li><strong>spiralize</strong>:</li>
</ul>
<blockquote class="blockquote">
<p><a href="https://doi.org/10.1093/bioinformatics/btab778" style="color: #758b76; font-weight: bold;">Zuguang Gu, et al., spiralize: an R package for Visualizing Data on Spirals, Bioinformatics, 2021.</a></p>
</blockquote>
<ul>
<li><strong>ComplexHeatmap</strong>:</li>
</ul>
<blockquote class="blockquote">
<p><a href="https://academic.oup.com/bioinformatics/article/32/18/2847/1743594" style="color: #758b76; font-weight: bold;">Complex heatmaps reveal patterns and correlations in multidimensional genomic data</a></p>
<p><a href="https://onlinelibrary.wiley.com/doi/10.1002/imt2.43" style="color: #758b76; font-weight: bold;">Complex heatmap visualization</a></p>
</blockquote>
<ul>
<li><strong>ggplot2</strong>:</li>
</ul>
<blockquote class="blockquote">
<p><a href="https://github.com/tidyverse/ggplot2" style="color: #758b76; font-weight: bold;">github:ggplot2</a></p>
<p><a href="https://ggplot2.tidyverse.org/" style="color: #758b76; font-weight: bold;">An implementation of the Grammar of Graphics in R</a></p>
</blockquote>
</blockquote>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="SSSYDYSSS/TransProRBook" data-repo-id="R_kgDOKiXTww" data-category="General" data-category-id="DIC_kwDOKiXTw84CaRd5" data-mapping="title" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" async="">
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Enrichment ggtree.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Enrichment ggtree</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Enrichment ggraph.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Enrichment ggraph</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>