# Data Source and Introduction {.unnumbered}
## Characteristics of the TCGA Database:

- Rich Cancer Tissue Sample Data:

> The TCGA (The Cancer Genome Atlas) database amasses a substantial volume of tissue sample data across a variety of cancer types, encompassing multi-dimensional data such as genomics, transcriptomics, and proteomics. This repository serves as a valuable resource for cancer research, enabling researchers to delve into the biological mechanisms of cancer at multiple levels.
  
- Primary Focus on Cancer Types:

> The core objective of TCGA is to unveil the molecular mechanisms, pathogenic genes, and mutation frequencies of different cancer types through integrative analysis, thereby providing a robust scientific basis for the diagnosis, treatment, and prevention of cancer.
  
- Limited Quantity of Normal Tissue Samples:

> Despite the richness of cancer sample data, TCGA has a relatively limited number of normal tissue samples, which to some extent, restricts its application in comparative studies between normal and cancerous tissues.
  
## Characteristics of the GTEx Database:

- Covers Normal Tissue Sample Data from Multiple Organs and Tissues:

> The GTEx (Genotype-Tissue Expression) database, by collecting samples from a variety of organs and tissues, offers an extensive gene expression data resource. This is crucial for understanding the heterogeneity of gene expression across different tissues and the functionality of genes under normal physiological conditions.

- Focus on Gene Expression Patterns in Normal Tissues:

> GTEx aims to reveal the patterns and variations of gene expression in normal tissues, providing a valuable foundation for researching gene functionality and regulatory mechanisms.
  
## Demands and Challenges of Joint Analysis:

- Need to Augment Normal Tissue Sample Volume:

> Given the limited number of normal tissue samples in the TCGA database, it's essential to source additional normal tissue samples from other databases like GTEx, for a more comprehensive and accurate cancer research.
  
- Complexity of Data Integration:

> The joint analysis of the TCGA and GTEx databases entails data integration and standardization, requiring a meticulous and scientific approach to ensure the efficacy of data integration and the accuracy of analysis results. This may include data cleansing, standardization, and a unified analysis workflow to ensure that data obtained from different databases can be accurately and effectively combined for analysis.


# Data Download and Processing Statement

## UCSC Xena Data Download pipelineï¼š

> - Log in to the official website: <a href="http://xena.ucsc.edu/" style="color: #6a926b; font-style: italic; font-weight: bold;">http://xena.ucsc.edu/</a>
> - Click "Launch Xena" to enter the main interface: <a href="https://xenabrowser.net/" style="color: #6a926b; font-style: italic; font-weight: bold;">https://xenabrowser.net/</a>
> - Click "DATA SETS" to enter the dataset interface: <a href="https://xenabrowser.net/datapages/" style="color: #6a926b; font-style: italic; font-weight: bold;">https://xenabrowser.net/datapages/</a>
> - TCGA and GTEX Data Download:
>   - <a href="https://xenabrowser.net/datapages/?cohort=GDC%20TCGA%20Breast%20Cancer%20(BRCA)&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">GDC TCGA Breast Cancer (BRCA) (20 datasets)</a>
>     - <a href="https://xenabrowser.net/datapages/?dataset=TCGA-BRCA.htseq_counts.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">HTSeq - Counts (n=1,217) GDC Hub</a>   
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-BRCA.htseq_counts.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/gencode.v22.annotation.gene.probeMap; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">ID/Gene Mapping (Download link, copyable)</a>
>
>     - <a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.GDC_phenotype.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">Phenotype (n=1,284) GDC Hub</a>   
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-BRCA.GDC_phenotype.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
>
>     - <a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.survival.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">survival data (n=1,260) GDC Hub</a>   
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-BRCA.survival.tsv; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
> 
>   - <a href="https://xenabrowser.net/datapages/?cohort=GDC%20TCGA%20Melanoma%20(SKCM)&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">GDC TCGA Melanoma (SKCM) (14 datasets)</a>
>     - <a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.htseq_counts.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">HTSeq - Counts (n=472) GDC Hub</a>   
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-SKCM.htseq_counts.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/gencode.v22.annotation.gene.probeMap; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">ID/Gene Mapping (Download link, copyable)</a> 
>   
>     - <a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.GDC_phenotype.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">Phenotype (n=477) GDC Hub</a>   
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-SKCM.GDC_phenotype.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
>
>     - <a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.survival.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">survival data (n=463) GDC Hub</a>   
>       - <a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-SKCM.survival.tsv; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
>   
>   - <a href="https://xenabrowser.net/datapages/?cohort=GTEX&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">GTEX</a>
>     - <a href="https://xenabrowser.net/datapages/?dataset=gtex_gene_expected_count&host=https%3A%2F%2Ftoil.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">TOIL RSEM expected_count (n=7,845) UCSC Toil RNA-seq Recompute</a>   
>       - <a href="https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/gtex_gene_expected_count.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 
>       - <a href="https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/probeMap%2Fgencode.v23.annotation.gene.probemap; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">ID/Gene Mapping (Download link, copyable)</a>        
> 
>     - <a href="https://xenabrowser.net/datapages/?dataset=GTEX_phenotype&host=https%3A%2F%2Ftoil.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">GTEX phenotype (n=9,783) UCSC Toil RNA-seq Recompute</a>   
>       - <a href="https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/GTEX_phenotype.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a> 

## Data Type

### count 

In the context of RNA-seq data analysis, "Count" is an important concept. It represents the number of times each gene or transcript is measured in an RNA-seq experiment. Typically, these counts are obtained through sequencing machines, which measure the number of fragments associated with each gene or transcript, also known as "reads count" or "fragment count," which indicates how many times each gene or transcript is covered or mapped in the sequencing data. These count data serve as the starting point for RNA-seq data analysis, providing raw information about the expression of genes or transcripts in different samples.

#### Use Cases:

>- Gene Expression Analysis: 
Count data can be used to compare the relative expression levels of different genes. By tallying the counts for each gene, we can determine which genes are expressed more or less in various samples, identifying key genes.
>- Differential Expression Analysis: 
Differential expression analysis aims to study changes in gene expression levels between different conditions or groups. In this context, count data is employed to compare gene expression levels under different conditions, often requiring statistical models to identify genes with significant differences.
>- Clustering Analysis: 
Count data can also be used for clustering analysis to identify groups of genes or samples with similar expression patterns. Through clustering analysis, genes with similar expression patterns can be grouped together.
>- Gene Annotation: 
Count data helps determine which genes or transcripts have evidence of measurement in the sequencing data, which is crucial for gene annotation.

#### Advantages:

>- Intuitiveness: 
Count data is obtained directly from RNA-seq experiments, providing an intuitive and straightforward representation of gene expression.
>- Flexibility: 
Count data is versatile and applicable to various RNA-seq data analysis tasks, including basic gene expression analysis and more complex differential expression analysis.

#### Disadvantages:

>- Impact of Sequencing Depth and Gene Length: 
Count data is influenced by sequencing depth and gene length, necessitating standardization when comparing expression levels between different samples or genes.
>- Zero Inflation: 
For lowly expressed genes, count data may exhibit zero inflation, where counts are predominantly zero. Specialized statistical methods are required to address this issue.
>- Resource-Intensive: 
Handling raw count data can demand significant computational resources and storage space, particularly for large-scale RNA-seq datasets.

In summary, count data plays a pivotal role in RNA-seq data analysis but typically requires further standardization and analysis to derive meaningful results. It is suitable for use with RNA-seq data types, particularly for gene expression analysis and differential expression analysis.

### Additional Recommendationsï¼š

In the TransPro data analysis suite, each R package (e.g., TransProR) and Python package (e.g., TransProPy) involves a vast array of mathematical algorithms, integrated machine learning techniques, and various deep learning modules. Therefore, our initial design's focus for data selection was on the following aspects: <b style="color: #6a926b;">ensuring a sufficiently large volume of data, comprehensive and diversified annotation files, balanced data distribution, and the feasibility of cross-database data integration</b>. However, please note that to ensure the reproducibility and robustness of our technology, we ultimately chose publicly available data from TCGA and CTEXâ€”specifically, BRCA data (large volume, publicly available, well-annotated) and SKCM data (fairly large volume, public, and balanced data distribution) as our initial data sets. We then performed a series of algorithmic explorations to uncover data features and designed visual representations of the results.

Although we have batch-normalized all the data, please be aware that users can directly analyze their test data, which is acceptable. However, if you intend to merge public data from major databases, please carefully review the data processing procedures and quality control metrics of different databases in advance (for the generation of TCGA and GTEX data, please see the reference)! We <b style="color: #6a926b;">recommend</b> conducting combined analyses only under <b style="color: #6a926b;">relatively similar conditions</b>. We <b style="color: #6a926b;">further suggest obtaining the original fastq files</b> from the databases. This way, TransPro users can generate count data with uniform methods and parameters, significantly minimizing the noise caused by batch effects and enhancing the accuracy and authenticity of the results.

# Reference

## TCGA Count Data
>   - <a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/" style="color: #758b76; font-weight: bold;">STAR Alignment and Counting</a>: The primary counting data in TCGA is generated by STAR, which provides gene ID, unstranded, and stranded counts data following alignment.
>   - <a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga" style="color: #758b76; font-weight: bold;">Multi-dimensional Data Collection</a>: TCGA collects many types of data including molecular characterization data for over 20,000 tumor and normal samples, contributing to the count data.

## GTEx Expected Count Data
>   - <a href="https://github.com/broadinstitute/gtex-pipeline" style="color: #758b76; font-weight: bold;">Pipeline Components for RNA-seq Alignment, Quantification</a>: GTEx utilizes a pipeline for RNA-seq alignment, quantification, and quality control among other analysis tasks, which lead to the generation of expected count data.
>   - <a href="https://hub.docker.com/r/broadinstitute/gtex_rnaseq" style="color: #758b76; font-weight: bold;">Alignment and Expression Quantification</a>: GTEx RNA-seq pipeline involves alignment (STAR), QC (RNA-SeQC), and expression quantification (RSEM & RNA-SeQC) for generating expected count data.
>   - <a href="https://www.nature.com/articles/ng.2653" style="color: #758b76; font-weight: bold;">76-base, Paired-end Illumina TruSeq RNA Protocol</a>: GTEx uses a 76-base, paired-end Illumina TruSeq RNA protocol, averaging around 50 million aligned reads per sample for RNA sequencing.

