<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="TransProR User Manual.">

<title>TransProRBook - 1&nbsp; Data Source and Introduction</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./summary.html" rel="next">
<link href="./index.html" rel="prev">
<link href="./TransProR_rlogo.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="custom.css">
<meta property="og:title" content="TransProRBook - 1&nbsp; Data Source and Introduction">
<meta property="og:description" content="R package TransProR.">
<meta property="og:image" content="https://github.com/SSSYDYSSS/TransProRBook/blob/main/TransProR_rlogo.png">
<meta property="og:site-name" content="TransProRBook">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Data Source and Introduction</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./TransProR_rlogo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">TransProRBook</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/SSSYDYSSS/TransProRBook" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <div class="dropdown">
      <a href="" title="Share" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Share"><i class="bi bi-share"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
              <i class="bi bi-bi-twitter pe-1"></i>
            Twitter
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
              <i class="bi bi-bi-facebook pe-1"></i>
            Facebook
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
              <i class="bi bi-bi-linkedin pe-1"></i>
            LinkedIn
            </a>
          </li>
      </ul>
    </div>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Data Source and Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pipeline.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">pipeline</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#characteristics-of-the-tcga-database" id="toc-characteristics-of-the-tcga-database" class="nav-link active" data-scroll-target="#characteristics-of-the-tcga-database"><span class="header-section-number">1.1</span> Characteristics of the TCGA Database:</a></li>
  <li><a href="#characteristics-of-the-gtex-database" id="toc-characteristics-of-the-gtex-database" class="nav-link" data-scroll-target="#characteristics-of-the-gtex-database"><span class="header-section-number">1.2</span> Characteristics of the GTEx Database:</a></li>
  <li><a href="#demands-and-challenges-of-joint-analysis" id="toc-demands-and-challenges-of-joint-analysis" class="nav-link" data-scroll-target="#demands-and-challenges-of-joint-analysis"><span class="header-section-number">1.3</span> Demands and Challenges of Joint Analysis:</a></li>
  <li><a href="#data-download-and-processing-statement" id="toc-data-download-and-processing-statement" class="nav-link" data-scroll-target="#data-download-and-processing-statement"><span class="header-section-number">2</span> Data Download and Processing Statement</a>
  <ul class="collapse">
  <li><a href="#ucsc-xena-data-download-pipeline" id="toc-ucsc-xena-data-download-pipeline" class="nav-link" data-scroll-target="#ucsc-xena-data-download-pipeline"><span class="header-section-number">2.1</span> UCSC Xena Data Download pipeline：</a></li>
  <li><a href="#data-type" id="toc-data-type" class="nav-link" data-scroll-target="#data-type"><span class="header-section-number">2.2</span> Data Type</a>
  <ul class="collapse">
  <li><a href="#count" id="toc-count" class="nav-link" data-scroll-target="#count"><span class="header-section-number">2.2.1</span> count</a></li>
  <li><a href="#rpkm" id="toc-rpkm" class="nav-link" data-scroll-target="#rpkm"><span class="header-section-number">2.2.2</span> RPKM</a></li>
  <li><a href="#advantages-1" id="toc-advantages-1" class="nav-link" data-scroll-target="#advantages-1"><span class="header-section-number">2.2.3</span> Advantages:</a></li>
  <li><a href="#fpkm" id="toc-fpkm" class="nav-link" data-scroll-target="#fpkm"><span class="header-section-number">2.2.4</span> FPKM</a></li>
  <li><a href="#tpm" id="toc-tpm" class="nav-link" data-scroll-target="#tpm"><span class="header-section-number">2.2.5</span> TPM</a></li>
  <li><a href="#cpm" id="toc-cpm" class="nav-link" data-scroll-target="#cpm"><span class="header-section-number">2.2.6</span> CPM</a></li>
  <li><a href="#rsem" id="toc-rsem" class="nav-link" data-scroll-target="#rsem"><span class="header-section-number">2.2.7</span> RSEM</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#additional-recommendations" id="toc-additional-recommendations" class="nav-link" data-scroll-target="#additional-recommendations"><span class="header-section-number">3</span> Additional Recommendations：</a></li>
  <li><a href="#reference" id="toc-reference" class="nav-link" data-scroll-target="#reference"><span class="header-section-number">4</span> Reference</a>
  <ul class="collapse">
  <li><a href="#tcga-count-data" id="toc-tcga-count-data" class="nav-link" data-scroll-target="#tcga-count-data"><span class="header-section-number">4.1</span> TCGA Count Data</a></li>
  <li><a href="#gtex-expected-count-data" id="toc-gtex-expected-count-data" class="nav-link" data-scroll-target="#gtex-expected-count-data"><span class="header-section-number">4.2</span> GTEx Expected Count Data</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/SSSYDYSSS/TransProRBook/edit/main/intro.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/SSSYDYSSS/TransProRBook/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/SSSYDYSSS/TransProRBook/blob/main/intro.qmd" class="toc-action">View source</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Data Source and Introduction</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="characteristics-of-the-tcga-database" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="characteristics-of-the-tcga-database"><span class="header-section-number">1.1</span> Characteristics of the TCGA Database:</h2>
<ul>
<li>Rich Cancer Tissue Sample Data:</li>
</ul>
<blockquote class="blockquote">
<p>The TCGA (The Cancer Genome Atlas) database amasses a substantial volume of tissue sample data across a variety of cancer types, encompassing multi-dimensional data such as genomics, transcriptomics, and proteomics. This repository serves as a valuable resource for cancer research, enabling researchers to delve into the biological mechanisms of cancer at multiple levels.</p>
</blockquote>
<ul>
<li>Primary Focus on Cancer Types:</li>
</ul>
<blockquote class="blockquote">
<p>The core objective of TCGA is to unveil the molecular mechanisms, pathogenic genes, and mutation frequencies of different cancer types through integrative analysis, thereby providing a robust scientific basis for the diagnosis, treatment, and prevention of cancer.</p>
</blockquote>
<ul>
<li>Limited Quantity of Normal Tissue Samples:</li>
</ul>
<blockquote class="blockquote">
<p>Despite the richness of cancer sample data, TCGA has a relatively limited number of normal tissue samples, which to some extent, restricts its application in comparative studies between normal and cancerous tissues.</p>
</blockquote>
</section>
<section id="characteristics-of-the-gtex-database" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="characteristics-of-the-gtex-database"><span class="header-section-number">1.2</span> Characteristics of the GTEx Database:</h2>
<ul>
<li>Covers Normal Tissue Sample Data from Multiple Organs and Tissues:</li>
</ul>
<blockquote class="blockquote">
<p>The GTEx (Genotype-Tissue Expression) database, by collecting samples from a variety of organs and tissues, offers an extensive gene expression data resource. This is crucial for understanding the heterogeneity of gene expression across different tissues and the functionality of genes under normal physiological conditions.</p>
</blockquote>
<ul>
<li>Focus on Gene Expression Patterns in Normal Tissues:</li>
</ul>
<blockquote class="blockquote">
<p>GTEx aims to reveal the patterns and variations of gene expression in normal tissues, providing a valuable foundation for researching gene functionality and regulatory mechanisms.</p>
</blockquote>
</section>
<section id="demands-and-challenges-of-joint-analysis" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="demands-and-challenges-of-joint-analysis"><span class="header-section-number">1.3</span> Demands and Challenges of Joint Analysis:</h2>
<ul>
<li>Need to Augment Normal Tissue Sample Volume:</li>
</ul>
<blockquote class="blockquote">
<p>Given the limited number of normal tissue samples in the TCGA database, it’s essential to source additional normal tissue samples from other databases like GTEx, for a more comprehensive and accurate cancer research.</p>
</blockquote>
<ul>
<li>Complexity of Data Integration:</li>
</ul>
<blockquote class="blockquote">
<p>The joint analysis of the TCGA and GTEx databases entails data integration and standardization, requiring a meticulous and scientific approach to ensure the efficacy of data integration and the accuracy of analysis results. This may include data cleansing, standardization, and a unified analysis workflow to ensure that data obtained from different databases can be accurately and effectively combined for analysis.</p>
</blockquote>
</section>
<section id="data-download-and-processing-statement" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data Download and Processing Statement</h1>
<section id="ucsc-xena-data-download-pipeline" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="ucsc-xena-data-download-pipeline"><span class="header-section-number">2.1</span> UCSC Xena Data Download pipeline：</h2>
<blockquote class="blockquote">
<ul>
<li>Log in to the official website: <a href="http://xena.ucsc.edu/" style="color: #6a926b; font-style: italic; font-weight: bold;">http://xena.ucsc.edu/</a></li>
<li>Click “Launch Xena” to enter the main interface: <a href="https://xenabrowser.net/" style="color: #6a926b; font-style: italic; font-weight: bold;">https://xenabrowser.net/</a></li>
<li>Click “DATA SETS” to enter the dataset interface: <a href="https://xenabrowser.net/datapages/" style="color: #6a926b; font-style: italic; font-weight: bold;">https://xenabrowser.net/datapages/</a></li>
<li>TCGA and GTEX Data Download:
<ul>
<li><a href="https://xenabrowser.net/datapages/?cohort=GDC%20TCGA%20Breast%20Cancer%20(BRCA)&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">GDC TCGA Breast Cancer (BRCA) (20 datasets)</a>
<ul>
<li><a href="https://xenabrowser.net/datapages/?dataset=TCGA-BRCA.htseq_counts.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">HTSeq - Counts (n=1,217) GDC Hub</a>
<ul>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-BRCA.htseq_counts.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/gencode.v22.annotation.gene.probeMap; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">ID/Gene Mapping (Download link, copyable)</a></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.GDC_phenotype.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">Phenotype (n=1,284) GDC Hub</a>
<ul>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-BRCA.GDC_phenotype.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.survival.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">survival data (n=1,260) GDC Hub</a>
<ul>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-BRCA.survival.tsv; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
</ul></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?cohort=GDC%20TCGA%20Melanoma%20(SKCM)&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">GDC TCGA Melanoma (SKCM) (14 datasets)</a>
<ul>
<li><a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.htseq_counts.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">HTSeq - Counts (n=472) GDC Hub</a>
<ul>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-SKCM.htseq_counts.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/gencode.v22.annotation.gene.probeMap; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">ID/Gene Mapping (Download link, copyable)</a></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.GDC_phenotype.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">Phenotype (n=477) GDC Hub</a>
<ul>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-SKCM.GDC_phenotype.tsv.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?dataset=TCGA-SKCM.survival.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">survival data (n=463) GDC Hub</a>
<ul>
<li><a href="https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-SKCM.survival.tsv; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
</ul></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?cohort=GTEX&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">GTEX</a>
<ul>
<li><a href="https://xenabrowser.net/datapages/?dataset=gtex_gene_expected_count&amp;host=https%3A%2F%2Ftoil.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold;">TOIL RSEM expected_count (n=7,845) UCSC Toil RNA-seq Recompute</a>
<ul>
<li><a href="https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/gtex_gene_expected_count.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
<li><a href="https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/probeMap%2Fgencode.v23.annotation.gene.probemap; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">ID/Gene Mapping (Download link, copyable)</a></li>
</ul></li>
<li><a href="https://xenabrowser.net/datapages/?dataset=GTEX_phenotype&amp;host=https%3A%2F%2Ftoil.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" style="color: #6a926b; font-style: italic; font-weight: bold; ">GTEX phenotype (n=9,783) UCSC Toil RNA-seq Recompute</a>
<ul>
<li><a href="https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/GTEX_phenotype.gz; Full metadata" style="color: #6a926b; font-style: italic; font-weight: bold;">download (Download link, copyable)</a></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
</blockquote>
</section>
<section id="data-type" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="data-type"><span class="header-section-number">2.2</span> Data Type</h2>
<section id="count" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="count"><span class="header-section-number">2.2.1</span> count</h3>
<p>In the context of RNA-seq data analysis, “Count” is an important concept. It represents the number of times each gene or transcript is measured in an RNA-seq experiment. Typically, these counts are obtained through sequencing machines, which measure the number of fragments associated with each gene or transcript, also known as “reads count” or “fragment count,” which indicates how many times each gene or transcript is covered or mapped in the sequencing data. These count data serve as the starting point for RNA-seq data analysis, providing raw information about the expression of genes or transcripts in different samples.</p>
<section id="use-cases" class="level4" data-number="2.2.1.1">
<h4 data-number="2.2.1.1" class="anchored" data-anchor-id="use-cases"><span class="header-section-number">2.2.1.1</span> Use Cases:</h4>
<ul>
<li>Gene Expression Analysis:</li>
</ul>
<blockquote class="blockquote">
<p>Count data can be used to compare the relative expression levels of different genes. By tallying the counts for each gene, we can determine which genes are expressed more or less in various samples, identifying key genes.</p>
</blockquote>
<ul>
<li>Differential Expression Analysis:</li>
</ul>
<blockquote class="blockquote">
<p>Differential expression analysis aims to study changes in gene expression levels between different conditions or groups. In this context, count data is employed to compare gene expression levels under different conditions, often requiring statistical models to identify genes with significant differences.</p>
</blockquote>
<ul>
<li>Clustering Analysis:</li>
</ul>
<blockquote class="blockquote">
<p>Count data can also be used for clustering analysis to identify groups of genes or samples with similar expression patterns. Through clustering analysis, genes with similar expression patterns can be grouped together.</p>
</blockquote>
<ul>
<li>Gene Annotation:</li>
</ul>
<blockquote class="blockquote">
<p>Count data helps determine which genes or transcripts have evidence of measurement in the sequencing data, which is crucial for gene annotation.</p>
</blockquote>
</section>
<section id="advantages" class="level4" data-number="2.2.1.2">
<h4 data-number="2.2.1.2" class="anchored" data-anchor-id="advantages"><span class="header-section-number">2.2.1.2</span> Advantages:</h4>
<ul>
<li>Intuitiveness:</li>
</ul>
<blockquote class="blockquote">
<p>Count data is obtained directly from RNA-seq experiments, providing an intuitive and straightforward representation of gene expression.</p>
</blockquote>
<ul>
<li>Flexibility:</li>
</ul>
<blockquote class="blockquote">
<p>Count data is versatile and applicable to various RNA-seq data analysis tasks, including basic gene expression analysis and more complex differential expression analysis.</p>
</blockquote>
</section>
<section id="disadvantages" class="level4" data-number="2.2.1.3">
<h4 data-number="2.2.1.3" class="anchored" data-anchor-id="disadvantages"><span class="header-section-number">2.2.1.3</span> Disadvantages:</h4>
<ul>
<li>Impact of Sequencing Depth and Gene Length:</li>
</ul>
<blockquote class="blockquote">
<p>Count data is influenced by sequencing depth and gene length, necessitating standardization when comparing expression levels between different samples or genes.</p>
</blockquote>
<ul>
<li>Zero Inflation:</li>
</ul>
<blockquote class="blockquote">
<p>For lowly expressed genes, count data may exhibit zero inflation, where counts are predominantly zero. Specialized statistical methods are required to address this issue.</p>
</blockquote>
<ul>
<li>Resource-Intensive:</li>
</ul>
<blockquote class="blockquote">
<p>Handling raw count data can demand significant computational resources and storage space, particularly for large-scale RNA-seq datasets.</p>
</blockquote>
<p>In summary, count data plays a pivotal role in RNA-seq data analysis but typically requires further standardization and analysis to derive meaningful results. It is suitable for use with RNA-seq data types, particularly for gene expression analysis and differential expression analysis.</p>
</section>
</section>
<section id="rpkm" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="rpkm"><span class="header-section-number">2.2.2</span> RPKM</h3>
<p>RPKM (Reads Per Kilobase of transcript per Million mapped reads) is a method used for quantifying gene expression. This technique overcomes two major barriers: the differences in sequencing depth between samples and the comparison of genes of different lengths. Below is a deeper analysis of the concept, calculation, and application of RPKM, as well as some of its limitations.</p>
<section id="principle-and-calculation-of-rpkm" class="level4" data-number="2.2.2.1">
<h4 data-number="2.2.2.1" class="anchored" data-anchor-id="principle-and-calculation-of-rpkm"><span class="header-section-number">2.2.2.1</span> Principle and Calculation of RPKM:</h4>
<ul>
<li>Principle:</li>
</ul>
<blockquote class="blockquote">
<p>The RPKM method corrects the read data in sequencing experiments, making it unaffected by sequencing depth and gene length differences, thereby allowing for a more accurate comparison of gene expression levels under different conditions.</p>
</blockquote>
<ul>
<li><p>Calculation Formula: <span class="math display">\[
RPKM = \frac{10^9 \times C}{N \times L}
\]</span></p></li>
<li><p>Where:</p></li>
</ul>
<blockquote class="blockquote">
<p><strong><em>C</em></strong> is the read count for a specific gene.</p>
<p><strong><em>N</em></strong> is the total number of mapped reads in the sequencing experiment.</p>
<p><strong><em>L</em></strong> is the transcript length (gene length in base pairs).</p>
<p>The <span class="math inline">\(\boldsymbol{10^9}\)</span> in the formula is a standardization factor used to adjust the units, making the results more comprehensible.</p>
</blockquote>
</section>
<section id="application-scenarios" class="level4" data-number="2.2.2.2">
<h4 data-number="2.2.2.2" class="anchored" data-anchor-id="application-scenarios"><span class="header-section-number">2.2.2.2</span> Application Scenarios:</h4>
<ul>
<li>Transcriptomic Studies:</li>
</ul>
<blockquote class="blockquote">
<p>In transcriptome sequencing (RNA-seq), researchers often use RPKM to quantify gene expression, especially when comparing samples from different species, different tissues, or under different treatment conditions.</p>
</blockquote>
<ul>
<li>Quantitative Studies of Gene Expression:</li>
</ul>
<blockquote class="blockquote">
<p>RPKM provides a way to standardize gene expression data, allowing data obtained from different experiments or under different conditions to be compared.</p>
</blockquote>
</section>
</section>
<section id="advantages-1" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="advantages-1"><span class="header-section-number">2.2.3</span> Advantages:</h3>
<ul>
<li>Correction for Gene Length:</li>
</ul>
<blockquote class="blockquote">
<p>RPKM takes into account the length of the genes, allowing for a fair comparison of expression levels between long and short genes, resolving the issue of read count bias. - Normalization for Sequencing Depth:</p>
</blockquote>
<blockquote class="blockquote">
<p>By standardizing the total mapped reads, RPKM allows researchers to compare data from different sequencing experiments.</p>
</blockquote>
<section id="challenges" class="level4" data-number="2.2.3.1">
<h4 data-number="2.2.3.1" class="anchored" data-anchor-id="challenges"><span class="header-section-number">2.2.3.1</span> Challenges:</h4>
<ul>
<li>Bias in Low-Abundance Transcripts:</li>
</ul>
<blockquote class="blockquote">
<p>RPKM may overestimate the expression levels for genes with low expression because their expression could be amplified due to statistical fluctuations or limitations in sequencing depth.</p>
</blockquote>
<ul>
<li>No Correction for Technical Variations:</li>
</ul>
<blockquote class="blockquote">
<p>RPKM does not correct for technical biases that may occur during library preparation or sequencing, and additional statistical methods may be required to correct these biases.</p>
</blockquote>
<ul>
<li>Not Suitable for Transcript Diversity:</li>
</ul>
<blockquote class="blockquote">
<p>If a gene has multiple isoforms, RPKM calculation may not reflect the true expression status of each transcript variant.</p>
</blockquote>
<p>Therefore, while RPKM is an important tool in RNA-seq data analysis, researchers need to consider its limitations when using it and consider employing other complementary methods, such as TPM (Transcripts Per Million) or more advanced quantification methods like model-based methods (e.g., DESeq2 or edgeR), which can handle differences and variations in data under different samples and conditions more effectively.</p>
</section>
</section>
<section id="fpkm" class="level3" data-number="2.2.4">
<h3 data-number="2.2.4" class="anchored" data-anchor-id="fpkm"><span class="header-section-number">2.2.4</span> FPKM</h3>
<p>FPKM (Fragments Per Kilobase of transcript, per Million mapped fragments) is an enhancement over RPKM, accounting for the number of sequenced fragments instead of just read counts. This modification is crucial, particularly when a single read might encompass multiple fragments due to events like “split reads” or “spliced reads” in sequencing data.</p>
<section id="fpkm-calculation" class="level4" data-number="2.2.4.1">
<h4 data-number="2.2.4.1" class="anchored" data-anchor-id="fpkm-calculation"><span class="header-section-number">2.2.4.1</span> FPKM Calculation:</h4>
<ul>
<li><p>The formula for calculating FPKM is: <span class="math display">\[
FPKM = \frac{\text{Number of fragments mapped to a gene}}{(\frac{\text{Total mapped fragments}}{1,000,000}) \times (\frac{\text{Length of the gene}}{1,000})}
\]</span></p></li>
<li><p>Where:</p></li>
</ul>
<blockquote class="blockquote">
<p><strong><em>“Fragment count of gene”</em></strong> refers to the number of fragments for a specific gene.</p>
<p><strong><em>“Gene length in kilobases”</em></strong> is the measurement of the gene length, using kilobases (thousands of base pairs) as the unit.</p>
<p><strong><em>“Total mapped fragments in millions”</em></strong> represents the summation of all fragments aligned with the reference genome, expressed in millions.</p>
</blockquote>
</section>
<section id="suitability" class="level4" data-number="2.2.4.2">
<h4 data-number="2.2.4.2" class="anchored" data-anchor-id="suitability"><span class="header-section-number">2.2.4.2</span> Suitability:</h4>
<p>FPKM is utilized in situations similar to RPKM, primarily in RNA-seq data analysis, especially when comparing expression levels across genes or between different samples. FPKM allows a more equitable comparison of gene expression by taking into account both gene length and sequencing depth.</p>
</section>
<section id="advantages-2" class="level4" data-number="2.2.4.3">
<h4 data-number="2.2.4.3" class="anchored" data-anchor-id="advantages-2"><span class="header-section-number">2.2.4.3</span> Advantages:</h4>
<blockquote class="blockquote">
<p>Recognizes gene length, making it suitable for comparing expression levels across genes of diverse lengths.By normalizing per million mapped fragments, FPKM enables more fair comparisons between different samples.</p>
</blockquote>
</section>
<section id="disadvantages-1" class="level4" data-number="2.2.4.4">
<h4 data-number="2.2.4.4" class="anchored" data-anchor-id="disadvantages-1"><span class="header-section-number">2.2.4.4</span> Disadvantages:</h4>
<blockquote class="blockquote">
<p>It may misrepresent low-abundance genes, as the total number of mapped fragments in the denominator could lead to inflated FPKM values for these genes.It doesn’t account for potential biases, such as discrepancies in sequencing depth, necessitating further adjustments.</p>
</blockquote>
</section>
<section id="rpkm-vs.-fpkm" class="level4" data-number="2.2.4.5">
<h4 data-number="2.2.4.5" class="anchored" data-anchor-id="rpkm-vs.-fpkm"><span class="header-section-number">2.2.4.5</span> RPKM vs.&nbsp;FPKM:</h4>
<p>Though they appear similar, the necessity for FPKM in addition to RPKM comes from their unique applications and methods of calculation.</p>
<ul>
<li><p>Differences:</p></li>
<li><p>Calculation Basis:</p></li>
</ul>
<blockquote class="blockquote">
<p>While RPKM normalizes gene expression levels based on read counts, gene length, and total reads, FPKM does so using fragment counts. This approach is essential for instances where a single read may correspond to multiple fragments.</p>
</blockquote>
<ul>
<li>Reads vs.&nbsp;Fragments:</li>
</ul>
<blockquote class="blockquote">
<p>RPKM, the earlier introduced method, accounts for read counts. FPKM, an advancement over RPKM, considers fragment counts, which is vital for scenarios involving “split reads” or “spliced reads.”</p>
</blockquote>
<ul>
<li>Nomenclature:</li>
</ul>
<blockquote class="blockquote">
<p>RPKM (Reads Per Kilobase of transcript, per Million mapped reads) emphasizes normalization using read counts, whereas FPKM highlights normalization based on fragment counts.</p>
</blockquote>
</section>
<section id="why-fpkm" class="level4" data-number="2.2.4.6">
<h4 data-number="2.2.4.6" class="anchored" data-anchor-id="why-fpkm"><span class="header-section-number">2.2.4.6</span> Why FPKM?</h4>
<blockquote class="blockquote">
<ul>
<li>RPKM is tailored for single-end sequencing, whereas FPKM is more applicable for paired-end RNA-seq because it takes into account that two reads can map to a single fragment, thus avoiding duplication in counting.</li>
<li>FPKM is generally preferred over RPKM as it offers more precision by including fragment counts, which is particularly important for genes undergoing splicing events.</li>
</ul>
</blockquote>
</section>
<section id="choosing-between-rpkm-and-fpkm" class="level4" data-number="2.2.4.7">
<h4 data-number="2.2.4.7" class="anchored" data-anchor-id="choosing-between-rpkm-and-fpkm"><span class="header-section-number">2.2.4.7</span> Choosing Between RPKM and FPKM:</h4>
<p>While FPKM is generally a better choice due to its enhanced accuracy, the advent of more modern methods like TPM (Transcripts Per Million) has overshadowed both RPKM and FPKM in RNA-seq data analysis. TPM normalizes expression levels based on every million mapped transcripts and is not influenced by gene length or sequencing depth, providing a more reliable reflection of relative expression levels.</p>
</section>
</section>
<section id="tpm" class="level3" data-number="2.2.5">
<h3 data-number="2.2.5" class="anchored" data-anchor-id="tpm"><span class="header-section-number">2.2.5</span> TPM</h3>
<p>TPM (Transcripts Per Million) is an accurate method for measuring gene expression levels, capable of eliminating the effects of sequencing depth and transcript length differences between various samples.</p>
<p>To calculate TPM, we first need to obtain the normalized read counts for each transcript, which are then standardized per million transcripts. The steps are as follows:</p>
<p>Calculate the read counts for each transcript, considering the effective length of the transcript (i.e., excluding the length of non-coding regions).</p>
<p>Compute the ratio of each transcript’s read count to its effective length, then multiply this ratio by a normalization factor to enable comparisons between samples.</p>
<section id="tpm-calculation-formula-is-as-follows" class="level4" data-number="2.2.5.1">
<h4 data-number="2.2.5.1" class="anchored" data-anchor-id="tpm-calculation-formula-is-as-follows"><span class="header-section-number">2.2.5.1</span> TPM calculation formula is as follows:</h4>
<p><span class="math display">\[
TPM_i = \left( \frac{x_i}{l_i} \right) \left( \frac{1}{\sum_j \frac{x_j}{l_j}} \right) 10^6
\]</span></p>
</section>
<section id="where" class="level4" data-number="2.2.5.2">
<h4 data-number="2.2.5.2" class="anchored" data-anchor-id="where"><span class="header-section-number">2.2.5.2</span> Where:</h4>
<blockquote class="blockquote">
<p><span class="math inline">\(\boldsymbol{x_i}\)</span> represents the read count of a specific transcript.</p>
<p><span class="math inline">\(\boldsymbol{l_i}\)</span> is the effective length of the transcript (in kilobases).</p>
<p>Subscripts <span class="math inline">\(\boldsymbol{i}\)</span> and <span class="math inline">\(\boldsymbol{j}\)</span> denote different transcripts.</p>
<p><span class="math inline">\(\boldsymbol{\sum_j \frac{x_j}{l_j}}\)</span> is the sum of all transcript read counts ratios to their effective lengths.</p>
</blockquote>
</section>
<section id="the-main-differences-between-tpm-and-rpkmfpkm" class="level4" data-number="2.2.5.3">
<h4 data-number="2.2.5.3" class="anchored" data-anchor-id="the-main-differences-between-tpm-and-rpkmfpkm"><span class="header-section-number">2.2.5.3</span> The main differences between TPM and RPKM/FPKM:</h4>
<ul>
<li>Calculation method:</li>
</ul>
<blockquote class="blockquote">
<p>With TPM, the expression level of each transcript is first normalized to the length of the transcript, then standardized to the total transcript expression in the sample. In contrast, RPKM/FPKM directly utilize the read counts of transcripts or fragments, normalizing based on transcript length and total read counts.</p>
</blockquote>
<ul>
<li>Normalization target:</li>
</ul>
<blockquote class="blockquote">
<p>After calculating TPM, the total of the TPM values for all transcripts will be 1,000,000, facilitating comparisons between different samples. The total for RPKM/FPKM is not consistent, as they are based on read counts.</p>
</blockquote>
<ul>
<li>Impact of gene length:</li>
</ul>
<blockquote class="blockquote">
<p>TPM normalizes the effect of gene length on expression level calculations by considering transcript length, making comparisons between transcripts of varying lengths more equitable. Although RPKM/FPKM also standardize based on length, they do not entirely eliminate the influence of gene length when comparing between samples.</p>
</blockquote>
</section>
<section id="the-preference-for-tpm-is-primarily-based-on-the-following-reasons" class="level4" data-number="2.2.5.4">
<h4 data-number="2.2.5.4" class="anchored" data-anchor-id="the-preference-for-tpm-is-primarily-based-on-the-following-reasons"><span class="header-section-number">2.2.5.4</span> The preference for TPM is primarily based on the following reasons:</h4>
<ul>
<li>Accuracy:</li>
</ul>
<blockquote class="blockquote">
<p>TPM more accurately reflects the actual expression levels of transcripts, without biases due to differences in sequencing depth or gene length between samples.</p>
</blockquote>
<ul>
<li>Comparability:</li>
</ul>
<blockquote class="blockquote">
<p>TPM values, after standardization, allow for direct comparisons of transcript expression levels between different samples and experiments.</p>
</blockquote>
<ul>
<li>Universality:</li>
</ul>
<blockquote class="blockquote">
<p>The TPM calculation method is suitable for various transcriptomic studies, including gene expression quantification, differential expression analysis, and more.</p>
</blockquote>
<p>In summary, while RPKM/FPKM remain useful in certain contexts, TPM is more widely accepted and utilized in modern transcriptomic analyses due to its accuracy and comparability.</p>
</section>
</section>
<section id="cpm" class="level3" data-number="2.2.6">
<h3 data-number="2.2.6" class="anchored" data-anchor-id="cpm"><span class="header-section-number">2.2.6</span> CPM</h3>
<p>CPM (Counts Per Million) is another prevalent normalization method for assessing gene or transcript expression levels, alongside other techniques such as RPKM, FPKM, and TPM. The principle behind CPM is to standardize the number of times a gene or transcript is measured (counts), allowing for the comparison of expression levels across different genes or transcripts while accounting for the total number of measurements. By normalizing each gene or transcript’s expression levels to per million measurements, CPM minimizes the impact of sequencing depth and differences between samples.</p>
<p>The calculation formula for CPM is as follows: <span class="math display">\[
CPM = \frac{\text{Counts of gene/transcript} \times 10^6}{\sum (\text{Counts of all genes/transcripts})}
\]</span></p>
<section id="where-1" class="level4" data-number="2.2.6.1">
<h4 data-number="2.2.6.1" class="anchored" data-anchor-id="where-1"><span class="header-section-number">2.2.6.1</span> Where:</h4>
<blockquote class="blockquote">
<p><strong><em>“Counts of gene/transcript”</em></strong> refers to the number of measurements (counts) for a particular gene or transcript.</p>
<p>“<span class="math inline">\(\boldsymbol{\sum}\)</span>” represents the summation symbol, calculating the total counts for all genes or transcripts.</p>
</blockquote>
</section>
<section id="cpm-is-applicable-for" class="level4" data-number="2.2.6.2">
<h4 data-number="2.2.6.2" class="anchored" data-anchor-id="cpm-is-applicable-for"><span class="header-section-number">2.2.6.2</span> CPM is applicable for:</h4>
<ul>
<li>Comparing relative expression levels of different genes or transcripts.</li>
<li>Comparing gene expression levels between samples, particularly when considering variations in sequencing depth and among samples.</li>
</ul>
</section>
<section id="advantages-3" class="level4" data-number="2.2.6.3">
<h4 data-number="2.2.6.3" class="anchored" data-anchor-id="advantages-3"><span class="header-section-number">2.2.6.3</span> Advantages:</h4>
<blockquote class="blockquote">
<ul>
<li><p>CPM is a straightforward and intuitive method of normalization, easy to understand and compute.</p></li>
<li><p>It’s suitable for comparisons between different samples, enabling a fairer assessment of gene expression differences.</p></li>
</ul>
</blockquote>
</section>
<section id="disadvantages-2" class="level4" data-number="2.2.6.4">
<h4 data-number="2.2.6.4" class="anchored" data-anchor-id="disadvantages-2"><span class="header-section-number">2.2.6.4</span> Disadvantages:</h4>
<blockquote class="blockquote">
<ul>
<li><p>CPM may encounter issues with lowly expressed genes because the denominator includes the total counts. For these genes, a smaller denominator might inflate the CPM value, leading to inaccuracies.</p></li>
<li><p>CPM does not account for factors like gene length and sequencing depth, potentially compromising accuracy in certain scenarios.</p></li>
</ul>
</blockquote>
<p>Despite its limitations, CPM stands as a relatively rapid and simplistic normalization technique, fitting for basic RNA-seq data analysis tasks. For instance, during differential analysis, it’s sometimes necessary to eliminate low-expression genes. Here, CPM calculations become relevant, aiding in the exclusion of these genes based on set criteria. For straightforward adjustments of counts, one can utilize the cpm() function within the edgeR package.</p>
</section>
</section>
<section id="rsem" class="level3" data-number="2.2.7">
<h3 data-number="2.2.7" class="anchored" data-anchor-id="rsem"><span class="header-section-number">2.2.7</span> RSEM</h3>
<p>Here, we delve further into the core principles and computational methods of RSEM (RNA-Seq by Expectation-Maximization). RSEM stands as a powerful tool for accurately estimating the expression levels of genes or transcripts. It is particularly proficient in handling RNA sequencing data, resolving issues of multimapping and accounting for sequence mismatches. At the heart of RSEM is the Expectation-Maximization (EM) algorithm, a statistical technique used for estimating the parameters of a probability model, especially in cases where the model involves latent variables that are not directly observable. By maximizing the log-likelihood function of the data, RSEM enables more accurate inference of the relative abundance of transcripts.</p>
<section id="the-computation-process-of-rsem" class="level4" data-number="2.2.7.1">
<h4 data-number="2.2.7.1" class="anchored" data-anchor-id="the-computation-process-of-rsem"><span class="header-section-number">2.2.7.1</span> The computation process of RSEM:</h4>
<ul>
<li>1.E-step (Expectation):</li>
</ul>
<blockquote class="blockquote">
<p>Utilizing the current estimates of the parameters, RSEM calculates the expected value of each transcript being sequenced. This step takes into account the sequencing error rate and the multimapping of reads.</p>
</blockquote>
<ul>
<li>2.M-step (Maximization):</li>
</ul>
<blockquote class="blockquote">
<p>Based on the data generated in the E-step, RSEM updates the parameter estimates to maximize the likelihood of the read data.</p>
</blockquote>
<p>Given the complexity of the calculations and models involved in RSEM, specialized software tools are usually employed for execution. These tools have built-in necessary mathematical models and numerical optimization methods, relieving the user from delving into the intricate details behind them.</p>
</section>
<section id="scope-of-application" class="level4" data-number="2.2.7.2">
<h4 data-number="2.2.7.2" class="anchored" data-anchor-id="scope-of-application"><span class="header-section-number">2.2.7.2</span> Scope of Application:</h4>
<blockquote class="blockquote">
<ul>
<li>High-precision estimation of transcript abundance, suitable for complex samples with a large number of genes and transcripts, and high heterogeneity.</li>
<li>Quantitative comparison of gene expression differences between different samples and studying variations between isoforms of transcripts.</li>
</ul>
</blockquote>
</section>
<section id="advantages-4" class="level4" data-number="2.2.7.3">
<h4 data-number="2.2.7.3" class="anchored" data-anchor-id="advantages-4"><span class="header-section-number">2.2.7.3</span> Advantages:</h4>
<blockquote class="blockquote">
<ul>
<li>RSEM accurately handles multimapped reads, offering a more precise expression level estimate compared to traditional methods.</li>
<li>Through sophisticated statistical models, RSEM fully considers the uncertainties of reads and potential errors during the sequencing process.</li>
</ul>
</blockquote>
</section>
<section id="disadvantages-3" class="level4" data-number="2.2.7.4">
<h4 data-number="2.2.7.4" class="anchored" data-anchor-id="disadvantages-3"><span class="header-section-number">2.2.7.4</span> Disadvantages:</h4>
<blockquote class="blockquote">
<ul>
<li>The RSEM calculation process is resource-intensive and time-consuming, potentially unsuitable for large-scale samples or rapid analysis workflows.</li>
<li>It requires high-quality data input and is sensitive to low-quality or highly biased data.</li>
</ul>
</blockquote>
<p>In practice, there’s an essential distinction between RSEM’s “expected_count” and the conventional raw “count.” The “expected_count” in RSEM is a statistically estimated floating-point number considering the uncertainties of read assignment, while traditional “count” is merely an integer read count based on raw sequencing data. This difference enables RSEM to provide more in-depth and accurate information on transcript abundance in certain scenarios.</p>
</section>
</section>
</section>
</section>
<section id="additional-recommendations" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Additional Recommendations：</h1>
<p>In the TransPro data analysis suite, each R package (e.g., TransProR) and Python package (e.g., TransProPy) involves a vast array of mathematical algorithms, integrated machine learning techniques, and various deep learning modules. Therefore, our initial design’s focus for data selection was on the following aspects: <b style="color: #6a926b;">ensuring a sufficiently large volume of data, comprehensive and diversified annotation files, balanced data distribution, and the feasibility of cross-database data integration</b>. However, please note that to ensure the reproducibility and robustness of our technology, we ultimately chose publicly available data from TCGA and CTEX—specifically, BRCA data (large volume, publicly available, well-annotated) and SKCM data (fairly large volume, public, and balanced data distribution) as our initial data sets. We then performed a series of algorithmic explorations to uncover data features and designed visual representations of the results.</p>
<p>Although we have batch-normalized all the data, please be aware that users can directly analyze their test data, which is acceptable. However, if you intend to merge public data from major databases, please carefully review the data processing procedures and quality control metrics of different databases in advance (for the generation of TCGA and GTEX data, please see the reference)! We <b style="color: #6a926b;">recommend</b> conducting combined analyses only under <b style="color: #6a926b;">relatively similar conditions</b>. We <b style="color: #6a926b;">further suggest obtaining the original fastq files</b> from the databases. This way, TransPro users can generate count data with uniform methods and parameters, significantly minimizing the noise caused by batch effects and enhancing the accuracy and authenticity of the results.</p>
</section>
<section id="reference" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Reference</h1>
<section id="tcga-count-data" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="tcga-count-data"><span class="header-section-number">4.1</span> TCGA Count Data</h2>
<blockquote class="blockquote">
<ul>
<li><a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/" style="color: #758b76; font-weight: bold;">STAR Alignment and Counting</a>: The primary counting data in TCGA is generated by STAR, which provides gene ID, unstranded, and stranded counts data following alignment.</li>
<li><a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga" style="color: #758b76; font-weight: bold;">Multi-dimensional Data Collection</a>: TCGA collects many types of data including molecular characterization data for over 20,000 tumor and normal samples, contributing to the count data.</li>
</ul>
</blockquote>
</section>
<section id="gtex-expected-count-data" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="gtex-expected-count-data"><span class="header-section-number">4.2</span> GTEx Expected Count Data</h2>
<blockquote class="blockquote">
<ul>
<li><a href="https://github.com/broadinstitute/gtex-pipeline" style="color: #758b76; font-weight: bold;">Pipeline Components for RNA-seq Alignment, Quantification</a>: GTEx utilizes a pipeline for RNA-seq alignment, quantification, and quality control among other analysis tasks, which lead to the generation of expected count data.</li>
<li><a href="https://hub.docker.com/r/broadinstitute/gtex_rnaseq" style="color: #758b76; font-weight: bold;">Alignment and Expression Quantification</a>: GTEx RNA-seq pipeline involves alignment (STAR), QC (RNA-SeQC), and expression quantification (RSEM &amp; RNA-SeQC) for generating expected count data.</li>
<li><a href="https://www.nature.com/articles/ng.2653" style="color: #758b76; font-weight: bold;">76-base, Paired-end Illumina TruSeq RNA Protocol</a>: GTEx uses a 76-base, paired-end Illumina TruSeq RNA protocol, averaging around 50 million aligned reads per sample for RNA sequencing.</li>
</ul>
</blockquote>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="SSSYDYSSS/TransProRBook" data-repo-id="R_kgDOKiXTww" data-category="General" data-category-id="DIC_kwDOKiXTw84CaRd5" data-mapping="title" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" async="">
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Preface</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./summary.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Summary</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>